### DISCUSSION TITLE
React Server Components and where NextJS fits in

### DISCUSSION THREAD

**Discussion by cyrus-za** [vercel/next.js#20384] [Ideas]
The react team [Announced](https://reactjs.org/blog/2020/12/21/data-fetching-with-react-server-components.html)  server components this week. Seeing as the nextjs team worked closely with the react team the last few months, I take it, this is old news to them, but to the rest of us, this raises a lot of questions. How does this fit in with next? Will next incorporate these server components into SSG, SSR, ISR etc? 

See this [Hacker News thread](https://news.ycombinator.com/item?id=25497065) where this exact topic is being discussed of next having SSR for years and react now changing from old traditional SSR support for frameworks like next and gatsby to native component level server rendering. 

I would like to keep using getStaticProps and getServerSideProps, but to additionally be able to do data fetching on child components rather than fetching it all on the page level and then passing down via props/context. This will simplify our codebase considerably, especially with apollo hooks. When getInitialProps got replaced by getStaticProps and getServerSideProps, the getDataFromTree method became unsupported by nextjs and the with-apollo example code changed considerably. If we can do data fetching on component level, we wont need that getDataFromTree method. 

There are many other advantages to this new server component idea, but it is not clear if/how next will suppor it and fit it in with the traditional SSG/SSR

**Comment by agos** [comment]
Guillermo Rauch [has tweeted](https://twitter.com/rauchg/status/1341082660488183808):

> React Server Components will be seamlessly integrated into Next.js, deployable to @vercel, backwards-compatible and incrementally adoptable.
